README.md

```diff
@@ -3,6 +3,36 @@ The dependencies and Dockerfile for creating the Marqo base image
 
 The marqo-base image is an image that has the necessary dependencies for the Marqo to be installed on and run. This speeds up the build process. 
 
+## Manage Dependencies
+
+The marqo-base image contains all the dependencies needed to run Marqo. We use `pip-compile` to 
+to manage the dependencies. It is a part of the `pip-tools` package and can be installed as
+```bash
+pip install pip-tools
+```
+In case you want to add a new dependency, you can add it to the `requirements.in` file and run
+```bash
+pip-compile requirements.in --output-file=requirements.txt --strip-extras
+```
+to generate the `requirements.txt` file. This file is used to install the dependencies in the Dockerfile.
+If any of the dependencies/sub-dependencies are not pinned and are updated when you run `pip-compile`, you should add
+the new version to the `requirements.in` file and run the above command again until they converge.
+We have an automated pipeline for this check. Ideally, you should not commit to the `requirements.txt` file directly.
```

Comment:
> Ideally, you should not commit to the `requirements.txt` file directly.
Is there any case we allow manual change of `requirements.txt` file? Show we rephrase this to be more affirmative? Like `You should not manually change 'requirements.txt' file.`

Reply:
We need to manually change the `requirements.txt` for cross-platform dependent packages. I will rephrase it and be more specific.